x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java

```diff
@@ -649,142 +711,284 @@ private void indexInvalidation(Collection<String> tokenIds, ActionListener<Token
+            } else {
+                final Map<String, Object> userTokenSource = (Map<String, Object>)
+                    ((Map<String, Object>) source.get("access_token")).get("user_token");
+                final String authString = (String) userTokenSource.get("authentication");
+                final Integer version = (Integer) userTokenSource.get("version");
+                final Map<String, Object> metadata = (Map<String, Object>) userTokenSource.get("metadata");
+                Version authVersion = Version.fromId(version);
+                try (StreamInput in = StreamInput.wrap(Base64.getDecoder().decode(authString))) {
+                    in.setVersion(authVersion);
+                    Authentication authentication = new Authentication(in);
+                    final String newUserTokenId = UUIDs.randomBase64UUID();
+                    final Instant refreshTime = clock.instant();
+                    Map<String, Object> updateMap = new HashMap<>();
+                    updateMap.put("refreshed", true);
+                    updateMap.put("refresh_time", refreshTime.toEpochMilli());
+                    updateMap.put("superseded_by", getTokenDocumentId(newUserTokenId));
+                    UpdateRequestBuilder updateRequest =
+                        client.prepareUpdate(SecurityIndexManager.SECURITY_INDEX_NAME, TYPE, tokenDocId)
+                            .setDoc("refresh_token", updateMap)
+                            .setFetchSource(true)
+                            .setRefreshPolicy(RefreshPolicy.IMMEDIATE);
+                    assert seqNo != SequenceNumbers.UNASSIGNED_SEQ_NO : "expected an assigned sequence number";
+                    updateRequest.setIfSeqNo(seqNo);
+                    assert primaryTerm != SequenceNumbers.UNASSIGNED_PRIMARY_TERM : "expected an assigned primary term";
+                    updateRequest.setIfPrimaryTerm(primaryTerm);
+                    executeAsyncWithOrigin(client.threadPool().getThreadContext(), SECURITY_ORIGIN, updateRequest.request(),
+                        ActionListener.<UpdateResponse>wrap(
+                            updateResponse -> {
+                                if (updateResponse.getResult() == DocWriteResponse.Result.UPDATED) {
+                                    logger.debug("updated the original token document to {}", updateResponse.getGetResult().sourceAsMap());
+                                    createUserToken(newUserTokenId, authentication, clientAuth, listener, metadata, true);
+                                } else if (backoff.hasNext()) {
+                                    logger.info("failed to update the original token document [{}], the update result was [{}]. Retrying",
+                                        tokenDocId, updateResponse.getResult());
+                                    client.threadPool().schedule(
+                                        () -> innerRefresh(
```

Comment:
I don't particularly like this format more, but it felt more readable than having only the two last parameters in a different line. I can definitely change this back

Reply:
I find it hard to read, maybe see what @bizybot @albertzaharovits or @tvernum think?

Reply:
I reverted, it felt strange the more I looked at it